r/src/inst/examples/NYC_flights_analysis_demo.R

```diff
@@ -0,0 +1,246 @@
+valid <- splits[[2]]
+test <- splits[[3]]
+
+# Run H2O's GLM
+# We only provide the required parameters, everything else is default
+glm <- h2o.glm(x = predictors, y = response, training_frame = train)
+
+# Show a detailed model summary
+glm
+
+# Get the rmse on the validation set
+h2o.rmse(h2o.performance(glm, newdata = valid)) 
+
+# The second model is another default glm, but trained on 80% of the data (here, we combine the training and validation splits to get more training data),
+# and cross-validated using 4 folds. Note that cross-validation takes longer and is not usually done for really large datasets.
+# Note:h2o.rbind makes a copy here, so it's better to use splitFrame with `ratios = c(0.8)` instead
+glm <- h2o.glm(x = predictors, y = response, training_frame = h2o.rbind(train, valid), nfolds = 4, seed = 0xDECAF)
+
+# Show a detailed summary of the cross validation metrics
+# This gives you an idea of the variance between the folds
+glm@model$cross_validation_metrics_summary
+
+# Get the cross-validated RMSE by scoring the combined holdout predictions.
+# (Instead of taking the average of the metrics across the folds
+h2o.rmse(h2o.performance(glm, xval = TRUE))
+
+# Make prediction on test set
+preds <- h2o.predict(glm, test)
+preds <- h2o.cbind(test,preds)
+
+# Convert back to Spark to utilize dplyr backend for aggregations that will later be used for visuals (Shiny app)
+pred_tbl <- hc$asSparkFrame(preds)
+
+# Create scored look up data for Shiny app
+lookup_tbl <- pred_tbl %>%
+  group_by(origin, dest, carrier, airline) %>%
+  summarize(
+    flights = n(),
+    distance = mean(distance),
+    avg_dep_delay = mean(dep_delay),
+    avg_arr_delay = mean(arr_delay),
+    avg_gain = mean(gain),
+    pred_gain = mean(predict)
+  )
+
+# Cache the look up table
+sdf_register(lookup_tbl, "lookup")
+tbl_cache(sc, "lookup")
+
+# Find distinct airport codes
+carrier_origin <- c("JFK", "LGA", "EWR")
+carrier_dest <- c("BOS", "DCA", "DEN", "HNL", "LAX", "SEA", "SFO", "STL")
+
+# Shiny UI
+ui <- fluidPage(theme = shinytheme("yeti"),
+  
+  # Set display mode to bottom
+  tags$script(' var setInitialCodePosition = function() 
+              { setCodePosition(false, false); }; '),
+  
+  # Title
+  titlePanel("NYCFlights13: Time Gained in Flight"),
+  
+  # Create sidebar 
+  sidebarLayout(
+    sidebarPanel(
+      radioButtons("origin", "Flight origin:",
+                   carrier_origin, selected = "JFK"),
+      br(),
+      
+      radioButtons("dest", "Flight destination:",
+                   carrier_dest, selected = "SFO")
+      
+    ),
+    
+    # Show a tabset that includes a plot, model, and table view
+    mainPanel(
+      tabsetPanel(type = "tabs", 
+                  tabPanel("Plot", plotOutput("plot")),
+                  tabPanel("Variable Importance", plotOutput("plotvarimp")),
+                  tabPanel("Data", DT::dataTableOutput("datatable")),
+                  tabPanel("Map", leafletOutput("map"))
+      )
+    )
+  )
```

Comment:
nit: is indentation correct?

Reply:
Good catch, let me check and reformat if needed :)

Reply:
Fixed, it wasn't good